From 4e4dfb701e7f1473cc6c8cecbd3e8eab49369fd8 Mon Sep 17 00:00:00 2001
From: Jonas Gorski <jonas.gorski@bisdn.de>
Date: Tue, 22 Nov 2022 13:48:48 +0100
Subject: [PATCH 14/16] modules: do not call hwmon_device_register_with_info()
 with NULL chip

hwmon_device_register_with_info() is supposed to require hwmon_chip_info
being passed, so passing NULL is not valid.

Recent kernels enforce this, but supplying valid hwmon_chip_info would
require rewriting the whole driver.

So instead revert to the old call and live with the warning.

// <smpl>
@@
expression ret;
expression E1,E2;
@@
-ret = hwmon_device_register_with_info(E1, E2, NULL, NULL, NULL);
+ret = hwmon_device_register(E1);
// </smpl>

Signed-off-by: Jonas Gorski <jonas.gorski@bisdn.de>
---
 packages/base/any/kernels/modules/accton_i2c_psu.c | 3 +--
 packages/base/any/kernels/modules/cpr_4011_4mxx.c  | 3 +--
 packages/base/any/kernels/modules/dps850.c         | 3 +--
 packages/base/any/kernels/modules/ym2651y.c        | 3 +--
 4 files changed, 4 insertions(+), 8 deletions(-)

diff --git a/packages/base/any/kernels/modules/accton_i2c_psu.c b/packages/base/any/kernels/modules/accton_i2c_psu.c
index 279ffd9d541c..3dcf9f51a623 100755
--- a/packages/base/any/kernels/modules/accton_i2c_psu.c
+++ b/packages/base/any/kernels/modules/accton_i2c_psu.c
@@ -333,8 +333,7 @@ static int accton_i2c_psu_probe(struct i2c_client *client,
     }
 
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(4,9,0)
-    data->hwmon_dev = hwmon_device_register_with_info(&client->dev, "accton_i2c_psu",
-                                                      NULL, NULL, NULL);
+    data->hwmon_dev = hwmon_device_register(&client->dev);
 #else
     data->hwmon_dev = hwmon_device_register(&client->dev);
 #endif
diff --git a/packages/base/any/kernels/modules/cpr_4011_4mxx.c b/packages/base/any/kernels/modules/cpr_4011_4mxx.c
index cfb58ee4d0cd..ee51c94bf352 100644
--- a/packages/base/any/kernels/modules/cpr_4011_4mxx.c
+++ b/packages/base/any/kernels/modules/cpr_4011_4mxx.c
@@ -246,8 +246,7 @@ static int cpr_4011_4mxx_probe(struct i2c_client *client,
     }
 
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(4,9,0)
-    data->hwmon_dev = hwmon_device_register_with_info(&client->dev, "cpr_4011_4mxx",
-                                                      NULL, NULL, NULL);
+    data->hwmon_dev = hwmon_device_register(&client->dev);
 #else
     data->hwmon_dev = hwmon_device_register(&client->dev);
 #endif
diff --git a/packages/base/any/kernels/modules/dps850.c b/packages/base/any/kernels/modules/dps850.c
index cb7c1d8f5560..8a3fd99d8c3a 100755
--- a/packages/base/any/kernels/modules/dps850.c
+++ b/packages/base/any/kernels/modules/dps850.c
@@ -256,8 +256,7 @@ static int dps850_probe(struct i2c_client *client,
 	}
 
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(4,9,0)
-    data->hwmon_dev = hwmon_device_register_with_info(&client->dev, "dps850",
-                                                      NULL, NULL, NULL);
+    data->hwmon_dev = hwmon_device_register(&client->dev);
 #else
 	data->hwmon_dev = hwmon_device_register(&client->dev);
 #endif
diff --git a/packages/base/any/kernels/modules/ym2651y.c b/packages/base/any/kernels/modules/ym2651y.c
index 1f641ab7a619..17dfb6f99af8 100755
--- a/packages/base/any/kernels/modules/ym2651y.c
+++ b/packages/base/any/kernels/modules/ym2651y.c
@@ -567,8 +567,7 @@ static int ym2651y_probe(struct i2c_client *client,
     }
 
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(4,9,0)
-    data->hwmon_dev = hwmon_device_register_with_info(&client->dev, "ym2651y",
-                                                      NULL, NULL, NULL);
+    data->hwmon_dev = hwmon_device_register(&client->dev);
 #else
     data->hwmon_dev = hwmon_device_register(&client->dev);
 #endif
-- 
2.42.0

